packages/markdown-it-myst/src/citations.ts

```diff
@@ -22,7 +25,7 @@ export interface Citation {
 export const citationsPlugin: PluginWithOptions = (md) => {
   const regexes = {
     citation: /^([^^-]|[^^].+?)?(-)?@([\w][\w:.#$%&\-+?<>~/]*)(.+)?$/,
-    inText: /^@([\w][\w:.#$%&\-+?<>~/]*)(\s*)(\[)?/,
+    inText: /^@([\w|{][\w:.#$%&\-+?<>~/]*[\w|}])(\s*)(\[)?/,
```

Comment:
Ah, ok, so this says the last character of the citation (besides whitespace) must be alphanumeric/underscore.

Definitely makes sense for punctuation, e.g. `@a&b.` leaving off the `.`. Something like `@a&b&` would leave of the last `&` - I don't know what sort of silly things are done with these citation keys, but I assume this is fine. Definitely an improvement to pulling in all punctuation to the key.

Reply:
Seems like if we really wanted to handle a single character citation key, we could probably expand this regex to have 2 cases, something like this... (if we are actually making this change, we would probably need to update some of the match group indexing...)

```suggestion
    inText: /^@(([\w|{][\w:.#$%&\-+?<>~/]*[\w|}])|\w)(\s*)(\[)?/,
```

Reply:
I think if there is a complex case like that suggesting `@{a&b&}` is better. I am updating to handle the single character -- using a non-capture group `(?:[\w])`.